apps/intellij/src/main/kotlin/dev/nx/console/toolWindow/NxToolWindowFactory.kt

```diff
@@ -0,0 +1,26 @@
+package dev.nx.console.toolWindow
+
+import com.intellij.openapi.components.service
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.wm.ToolWindow
+import com.intellij.openapi.wm.ToolWindowFactory
+import dev.nx.console.services.NxlsService
+import kotlinx.coroutines.runBlocking
+
+class NxToolWindowFactory : ToolWindowFactory {
+    override fun createToolWindowContent(project: Project, toolWindow: ToolWindow) {
+        val toolwindowPanel = NxToolWindowPanel(project)
+        val contentManager = toolWindow.contentManager
+        val content = contentManager.factory.createContent(toolwindowPanel, null, false)
+        content.isCloseable = false
+        contentManager.addContent(content)
+    }
+}
+
+fun Project.nxWorkspace() = runBlocking { service<NxlsService>().workspace() }
```

Comment:
I don't know about having an extension function here. For `nxBasePath` is makes sense because we always use it instead of `project.basePath`. But here we only hide the fact that we're calling the `NxlsService`. Imo it would be better to have it explicit but feel free to explain why you did it this way. @Cammisuli you can also weigh in.

Reply:
just found it an easy way to get the nxWorkspace without much code we can refactor it and inline it...

Reply:
yeah it is pretty short that's true